edx_rest_framework_extensions/utils.py

```diff
@@ -38,21 +54,33 @@ def jwt_decode_handler(token):
 
     options = {
         'verify_exp': api_settings.JWT_VERIFY_EXPIRATION,
+        'verify_aud': settings.JWT_AUTH['JWT_VERIFY_AUDIENCE'],
         'require_exp': True,
-        'require_iat': True,
+        'require_iat': True
     }
 
-    try:
-        return jwt.decode(
-            token,
-            api_settings.JWT_SECRET_KEY,
-            api_settings.JWT_VERIFY,
-            options=options,
-            leeway=api_settings.JWT_LEEWAY,
-            audience=api_settings.JWT_AUDIENCE,
-            issuer=api_settings.JWT_ISSUER,
-            algorithms=[api_settings.JWT_ALGORITHM]
-        )
-    except jwt.InvalidTokenError:
-        logger.exception('JWT decode failed!')
-        raise
+    # JWT_ISSUERS is not one of DRF-JWT's default settings, and cannot be accessed
+    # using the `api_settings` object without overriding DRF-JWT's defaults.
+    issuers = get_jwt_issuers()
+    secret_keys = get_jwt_secret_keys()
+
+    for secret_key in secret_keys:
```

Comment:
As I mentioned earlier, this is a bad implementation that should not be repeated. Instead of a nested for-loop, the appropriate data should be packaged in a single dict for each provider/issuer:

```python
JWT_ISSUERS = [
    {
        'ISSUER': 'https://example.com/auth',
        'SECRET_KEY': 'blah!',
    },
    {
        'ISSUER': 'https://acme.com/auth',
        'SECRET_KEY': 'foo',
    },
]
```